nntrainer/graph/network_graph.cpp

```diff
@@ -217,6 +232,20 @@ int NetworkGraph::realizeFlattenType(Layer &current) {
   layer->input_layers.clear();
   layer->input_layers.push_back(current.getName());
 
+  if (current.getNumOutputs() != 1)
+    return ML_ERROR_INVALID_PARAMETER;
+
+  layer->setNumOutputs(current.getNumOutputs());
+  layer->output_layers.clear();
+  for (unsigned int i = 0; i < current.getNumOutputs(); ++i)
+    layer->output_layers.push_back(current.output_layers[i]);
```

Comment:
I think flatten layer shouldn't have multi output.

Reply:
line 235 ensures that this loops runs only once.

Reply:
> line 235 ensures that this loop runs only once.

Do you mean current.getNumOutputs() is always 1? I think it is better to set the output_layer after all the input is set properly. It if is calculated layer by layer.. then there might be an error, I guess. Cause.. if the current number of output is multiple.. that means.. we needs multi-out layer.. so.. Could you check this?

Reply:
Yes, the current.getNumOutputs() must always be 1. Flatten is used only in this way.  If flatten has to be used with multi-output, then the flatten must be set before the multi-output. If the flatten is supposed to be used for a single output of the multi-output, then it should be inserted manually there inside the ini model.
If you still think there is a bug in the situation you described, can you please describe a minimal model showing the bug.